### DISCUSSION TITLE
Build as dll

### DISCUSSION THREAD

**Discussion by PharmCat** [InteractiveComputerGraphics/SPlisHSPlasH#132] [Q&A] [answered]
Hi! Is it possible to build SPlisHSPlasH as dll and call it to stars simulation from another project?

**Comment by janbender** [comment, marked as answer]
I never tried to build it as DLL. You can try to switch to "SHARED" in the CMake files to see if that works.

**Comment by PharmCat** [comment]
Thaks! 
I add SHARE and get:

```
Error	LNK2001	неразрешенный внешний символ "class Utilities::Logger Utilities::logger" (?logger@Utilities@@3VLogger@1@A).	SPlisHSPlasH	\SPlisHSPlasH\build\SPlisHSPlasH\SimpleQuadrature.obj	1	
```

```
LNK2019	ссылка на неразрешенный внешний символ "public: void __cdecl CompactNSearch::NeighborhoodSearch::resize_point_set(unsigned int,float const *,unsigned __int64)" (?resize_point_set@NeighborhoodSearch@CompactNSearch@@QEAAXIPEBM_K@Z) в функции "public: void __cdecl SPH::Emitter::emitParticles(class std::vector<unsigned int,class std::allocator<unsigned int> > &,unsigned int &,unsigned int &)" (?emitParticles@Emitter@SPH@@QEAAXAEAV?$vector@IV?$allocator@I@std@@@std@@AEAI1@Z).	SPlisHSPlasH	\SPlisHSPlasH\build\SPlisHSPlasH\Emitter.obj	1	
```
and others...

OK. I think there is no simple way. I try to find a way to use SPlisHSPlasH with Julia, so potentially shared library can be used with Julia with [CxxWrap.jl](https://github.com/JuliaInterop/CxxWrap.jl) or something else...

**Comment by digitalillusions** [comment]
It is not a problem to build as a shared library on linux. On windows however this is unfortunately not directly possible. See my answer here https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/issues/48#issuecomment-897452534 .

TL;DR on windows it's not directly possible to build splishsplash as a shared library because all symbols have to be declared for export "manually".
